### PR TITLE
chore(dgeni): update to latest dgeni and dgeni-packages

### PR DESCRIPTION
**Pull request by Splaktar**
# AngularJS is in LTS mode
We are no longer accepting changes that are not critical bug fixes into this project.
See https://blog.angular.io/stable-angularjs-and-long-term-support-7e077635ee9c for more detail.

**Does this PR fix a regression since 1.7.0, a security flaw, or a problem caused by a new browser version?**
Yes

**What is the current behavior? (You can also link to an open issue here)**
- from: 389 vulnerabilities found - Severity: 259 Low | 28 Moderate | 98 High | 4 Critical

**What is the new behavior (if this is a feature change)?**
- to: 300 vulnerabilities found - Severity: 189 Low | 28 Moderate | 79 High | 4 Critical

**Does this PR introduce a breaking change?**
No.

- ~~See the broken markdown rendering screen shots at the end~~
- ~~`errorOnUnmatchedLinks` used to pass but it no longer does, this is likely because of the
  failure of the markdown renderer to function~~

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated
- [x] Fix/Feature: Tests have been added; existing tests pass

**Other information**:

### Before
Screen Shot 2021-06-05 at 16 59 17

### After
Screen Shot 2021-06-05 at 16 50 52

This same problem was observed in AngularJS Material in PR https://github.com/angular/material/pull/11881.

### Fixed rendering in a 2nd commit

chore(dgeni): fix rendering of docs markdown

- this adds 1 High severity advisory back in by pinning at `marked@0.3.6`
- 301 vulnerabilities found - Severity: 189 Low | 28 Moderate | 80 High | 4 Critical
- in reality, there are many serious security issues with this old version of `marked`
- but our docs don't render with the newer versions due to
  https://github.com/angular/dgeni-packages/issues/310
- another downside is that this forces `firebase-tools` to use an old, vulnerable
  version of marked for parsing console output

### CODE REVIEW

**Comment by Splaktar** [comment]
This appears to be caused only by the update to `dgeni-packages@0.28.0`, which has breaking changes related to `marked`, as keeping `dgeni@0.4.9` still reproduces the issue.

**Comment by Splaktar** [comment]
I wrote up the issue that I was able to isolate a bit better via this PR in https://github.com/angular/dgeni-packages/issues/310.

**Comment by Splaktar** [comment]
I've opened https://github.com/markedjs/marked/issues/2085 for this as well.

**Comment by Splaktar** [comment]
Even with the concerns above, this now fixes 88 High and Low severity vulnerabilities without breaking rendering of markdown in the docs.

File: package.json [hunk 1]
```diff
@@ -106,7 +105,8 @@
     "//2": "(E.g. see https://github.com/gulpjs/gulp/issues/2162 and https://github.com/nodejs/node/issues/25132.)",
     "natives": "1.1.6",
     "//3": "`graceful-fs` needs to be pinned to support gulp 3, on Node v12+",
-    "graceful-fs": "^4.2.3"
+    "graceful-fs": "^4.2.3",
+    "marked": "0.3.6"
```
Comment (petebacondarwin) [inline package.json:109]: Rather than pinning `marked` we should just fix the docs to be CommonMark compliant.

Comment (Splaktar) [reply package.json:109]: OK, I wasn't sure that was something that we were open to doing. I'll look into it.

**Comment by Splaktar** [comment]
Closing in favor of https://github.com/angular/angular.js/issues/17163 at Pete's suggestion.

**Pull request closed without merging**